wls-briefwahl-service/src/main/java/de/muenchen/oss/wahllokalsystem/briefwahlservice/configuration/UserInfoAuthoritiesService.java

```diff
@@ -0,0 +1,121 @@
+        this.restTemplate = restTemplateBuilder.build();
+        this.cache = new CaffeineCache(NAME_AUTHENTICATION_CACHE,
+                Caffeine.newBuilder()
+                        .expireAfterWrite(AUTHENTICATION_CACHE_ENTRY_SECONDS_TO_EXPIRE, TimeUnit.SECONDS)
+                        .ticker(Ticker.systemTicker())
+                        .build());
+    }
+
+    /**
+     * Ruft den /userinfo Endpoint und extrahiert {@link GrantedAuthority}s aus dem "authorities"
+     * Claim.
+     *
+     * @param jwt der JWT
+     * @return die {@link GrantedAuthority}s gem. Claim "authorities" des /userinfo Endpoints
+     */
+    public Collection<SimpleGrantedAuthority> loadAuthorities(Jwt jwt) {
+        ValueWrapper valueWrapper = this.cache.get(jwt.getSubject());
+        if (valueWrapper != null) {
+            // value present in cache
+            @SuppressWarnings("unchecked")
+            Collection<SimpleGrantedAuthority> authorities = (Collection<SimpleGrantedAuthority>) valueWrapper.get();
+            log.debug("Resolved authorities (from cache): {}", authorities);
+            return authorities;
+        }
+
+        log.debug("Fetching user-info for token subject: {}", jwt.getSubject());
+        final HttpHeaders headers = new HttpHeaders();
+        headers.set(HttpHeaders.AUTHORIZATION, "Bearer " + jwt.getTokenValue());
+        final HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        Collection<SimpleGrantedAuthority> authorities = new ArrayList<>();
+        try {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> map = restTemplate.exchange(this.userInfoUri, HttpMethod.GET, entity,
+                    Map.class).getBody();
+
+            log.debug("Response from user-info Endpoint: {}", map);
+            if (map.containsKey(CLAIM_AUTHORITIES)) {
+                authorities = asAuthorities(map.get(CLAIM_AUTHORITIES));
+            }
+            log.debug("Resolved Authorities (from /userinfo Endpoint): {}", authorities);
+            // store
+            this.cache.put(jwt.getSubject(), authorities);
+        } catch (Exception e) {
+            log.error(String.format("Could not fetch user details from %s - user is granted NO authorities",
+                    this.userInfoUri), e);
+        }
+
+        return authorities;
+    }
+
+    private static List<SimpleGrantedAuthority> asAuthorities(Object object) {
+        List<SimpleGrantedAuthority> authorities = new ArrayList<>();
+        if (object instanceof Collection) {
+            Collection<?> collection = (Collection<?>) object;
+            object = collection.toArray(new Object[0]);
+        }
+        if (ObjectUtils.isArray(object)) {
+            authorities.addAll(
+                    Stream.of(((Object[]) object))
+                            .map(Object::toString)
+                            .map(SimpleGrantedAuthority::new)
+                            .collect(Collectors.toList()));
```

Comment:
```suggestion
                            .toList());
```
Macht er das so nicht auch?

Reply:
done mit 834a050d42777bccbd594f5c81592909e00be450